projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -56,9 +56,9 @@ import { MultiSelectJustify } from './multi-select-justify';
         </ht-popover-trigger>
         <ht-popover-content>
           <div class="multi-select-content" [ngStyle]="{ 'min-width.px': triggerContainer.offsetWidth }">
-            <ng-container *ngIf="this.enableSearch">
+            <ng-container *ngIf="this.searchMode !== '${MultiSelectSearchMode.Disabled}'">
               <ng-container *ngIf="this.allOptions$ | async as allOptions">
-                <ng-container *ngIf="allOptions.length > 5">
+                <ng-container *ngIf="this.searchMode === '${MultiSelectSearchMode.EmitOnly}' || allOptions.length > 5">
```

Comment:
this condition is non obvious to me - the two search modes don't really dictate when the search box should or should be shown - I'd think it would be either the same logic for both, or no logic and let the caller dictate by explicitly disabling filter if they're not providing 5+ options (can't remember where exactly that conversation landed in the previous PR).

Reply:
That problem comes up when you search and get less than 5 results. You are stuck at that point. We can't do it with an option, since this logic will not work with server side search.

Reply:
That's actually why I'd rather take the logic out entirely - if I express a search mode, show it. If I want to hide the search box in certain conditions, that's trivially done on the caller side.

Reply:
Cool. I will take it out

Reply:
The only thing is, for consistent behavior we will have to put `searchMode= options.length > 5 ? MultiSelectSearchMode.Implicit : MultiSelectSearchMode.Disabled` in every usage.  It looks really weird to show search with just 1 option.

Reply:
There's very places where we have a variable number of options, anyway - and I think with any static small number (such as 6) search doesn't really make sense.

Reply:
We need the options.length > 1 logic for displaying buttons 'select all' and 'clear selected' buttons. I can remove `allOptions.length > 5` conditions if you feel it is not useful and condition is non obvious.

Reply:
Removed